src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -239,7 +239,9 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
         ));
 
         $resolver->setAllowedTypes(array(
+            'choices' => array('null', 'array'),
```

Comment:
The OptionsResolver cannot make an option required only when the other is not there.